SteamBus.App/src/SteamBus.DBus/SteamClient.cs

```diff
@@ -476,42 +476,71 @@ Task IPluginLibraryProvider.RefreshAsync()
     return Task.FromResult(0);
   }
 
-  async Task<ItemMetadata> IPluginLibraryProvider.GetAppMetadataAsync(string appIdString)
+  async Task<String> IPluginLibraryProvider.GetItemMetadataAsync(string appIdString)
   {
     if (!await EnsureConnected()) throw DbusExceptionHelper.ThrowNotLoggedIn();
     if (ParseAppId(appIdString) is not uint appId) throw DbusExceptionHelper.ThrowInvalidAppId();
 
-    await session!.RequestAppInfo(appId, true);
+    var CommonSection = session!.GetSteam3AppSection(appId, EAppInfoSection.Common);
+    var name = CommonSection?["name"].Value?.ToString();
+    var app_type = CommonSection?["type"].Value?.ToString() ?? "";
+    var parent_store_id = CommonSection?["parent"].Value?.ToString() ?? "";
 
-    var info = depotConfigStore.GetInstalledAppInfo(appId);
-    var name = session!.GetSteam3AppName(appId);
-    var requiresInternetConnection = session!.GetSteam3AppRequiresInternetConnection(appId);
-
-    if (info == null)
+    if (name == null)
     {
-      return new ItemMetadata
-      {
-        Name = name,
-        InstallSize = 0,
-        RequiresInternetConnection = requiresInternetConnection,
-        CloudSaveFolders = [],
-        InstalledVersion = "",
-        LatestVersion = "",
-      };
+      return "";
+    }
+    if (
+      !string.Equals(app_type, "game", StringComparison.OrdinalIgnoreCase)
+      || !string.Equals(app_type, "demo", StringComparison.OrdinalIgnoreCase)
+      || !string.Equals(app_type, "beta", StringComparison.OrdinalIgnoreCase))
+    {
+      return "";
     }
 
-    var latestVersion = session!.GetSteam3AppBuildNumber(appId, info!.Value.Branch);
+    // TODO: Check for expired betas and demos
```

Comment:
We might want to handle this before merging

Reply:
Steam doesn't handle this very gracefully. Old open betas can be installed but all the content has been moved to a private branch so it gives a missing executable error on launch.

Let's not block this feature because of that